### PR TITLE
Add swift-server CI

### PR DESCRIPTION
**Pull request by fabianfett**
### Motivation

We want to use the swift-server CI going forward. For this we need to add a bunch of docker magic to work.

### Changes

- Add docker-compose files to run tests in swift-server ci

### Open ends

Some of the CI jobs will fail. We should enable it and one by one make it green. First one is already posted: #52.

### CODE REVIEW

**Comment by Joannis** [comment]
@fabianfett how do builds get triggered?

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@Joannis CI should be triggered automatically once I set them up, or they can be force-triggered using "test this please" comment (e.g., https://github.com/swift-server/RediStack/pull/54#issuecomment-1594935859)

**Comment by fabianfett** [comment]
CI fails as expected. Will fix with follow up PR #52.

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# tools
+RUN mkdir -p $HOME/.tools
+RUN echo 'export PATH="$HOME/.tools:$PATH"' >> $HOME/.profile
+
+# swiftformat (until part of the toolchain)
+
+ARG swiftformat_version=0.48.8
+RUN git clone --branch $swiftformat_version --depth 1 https://github.com/nicklockwood/SwiftFormat $HOME/.tools/swift-format
+RUN cd $HOME/.tools/swift-format && swift build -c release
+RUN ln -s $HOME/.tools/swift-format/.build/release/swiftformat $HOME/.tools/swiftformat
```
Comment (yim-lee) [inline docker/Dockerfile:34]: Can remove this block if there is no plan to run swiftformat.

**Review by yim-lee — APPROVED** [review]
CI jobs are configured. We can add soundness check job etc. later when ready. Just let me know.

**Review by Joannis — APPROVED** [review]

**Pull request merged**